src/ObjCRuntime/Blocks.cs

```diff
@@ -83,14 +83,19 @@ public unsafe struct BlockLiteral {
 		[DllImport ("__Internal")]
 		static extern IntPtr xamarin_get_block_descriptor ();
 
+		struct DelegatePairs {
+			public Delegate Trampoline, UserDelegate;
+		}
+
 		//
 		// trampoline must be static, and someone else needs to keep a ref to it
 		//
 		public unsafe void SetupBlock (Delegate trampoline, Delegate userDelegate)
 		{
 			isa = block_class;
 			invoke = Marshal.GetFunctionPointerForDelegate (trampoline);
-			local_handle = (IntPtr) GCHandle.Alloc (userDelegate);
+			var delegates = new DelegatePairs () { Trampoline = trampoline, UserDelegate = userDelegate };
```

Comment:
The struct will be boxed, so there's no memory saved by using a struct.

The object stored in `local_handle` is different now, which means that the code that fetches the delegate stored in `local_handle` (in line 160) needs to be updated too.